src/python/deepgnn/graph_engine/snark/converter/writers.py

```diff
@@ -93,6 +95,10 @@ def add(self, data: typing.Iterator[typing.Tuple[int, int, int, float, list]]):
                 self.node_type_count[typ] += 1
                 self.node_count += 1
             else:
+                if src != self.prev_node_id or self.prev_node_id is None:
+                    self.node_writer.add(src, -1, [])
```

Comment:
Please, update documentation, that -1 is the reserved node type and can't be used in a graph.